site/app/controllers/student/TeamController.php

```diff
@@ -497,6 +497,28 @@ public function setTeamName($gradeable_id): RedirectResponse {
             $_POST['team_name'] = null;
         }
 
+        if(strlen($_POST['team_name']) > 30) {
+            $this->core->addErrorMessage("Number of characters in team name must be less than 30");
+            return new RedirectResponse($return_url);
+        }
+        
+        if(preg_match('/[\x{1F600}-\x{1F64F}\x{1F300}-\x{1F5FF}\x{1F680}-\x{1F6FF}\x{1F700}-\x{1F77F}\x{1F780}-\x{1F7FF}\x{1F800}-\x{1F8FF}\x{1F900}-\x{1F9FF}\x{1FA00}-\x{1FA6F}\x{1FA70}-\x{1FAFF}\x{2600}-\x{26FF}\x{2700}-\x{27BF}\x{2300}-\x{23FF}]/u',$_POST['team_name'])) {
+            $this->core->addErrorMessage("Team name should not contain emojis");
+            return new RedirectResponse($return_url);
+        } 
+
+        function hasSpecialCharacters($str) {
```

Comment:
It would be good to move this function outside `setTeamName`, I think just leaving it in the same controllerFile and making it a `static` function would be good so you can call it from anywhere and it'll be easier to write a unit test for it